sierra_adapter/sierra_indexer/src/main/scala/weco/catalogue/sierra_indexer/services/Splitter.scala

```diff
@@ -36,10 +36,25 @@ class Splitter(indexPrefix: String) {
 
   private def getSierraApiData(t: SierraTransformable)
     : Either[Seq[(Parent, ParsingFailure)], Seq[(Parent, Json)]] = {
+    val itemIds = t.itemRecords.keys.map { _.withoutCheckDigit }.toList.sorted
+    val holdingsIds =
+      t.holdingsRecords.keys.map { _.withoutCheckDigit }.toList.sorted
+
     val bibData = t.maybeBibRecord match {
       case Some(bibRecord) =>
         Seq(
-          Parent(SierraRecordTypes.bibs, bibRecord.id) -> parse(bibRecord.data)
+          Parent(SierraRecordTypes.bibs, bibRecord.id) ->
```

Comment:
Could we use a case class to do this?

Reply:
Not really, or at least not without reharvesting all the data – I'll add a comment to clarify what it's doing here.

This app works mostly on `Json` instances, so we're passing around the data exactly as we get it from the Sierra API. If you turn that into a case class, you only get the fields you explicitly include on the case class.

This is injecting the "itemIds" and "holdingIds" fields alongside the bib data fields. It's slightly cheating, but such a useful piece of info I figured it worth adding. It mimics similar fields on the items/holdings records.